### ISSUE TITLE
Weird build error, "does not exist"

### ISSUE THREAD

**Issue by bitemyapp** [commercialhaskell/stack#976]
```
$ stack --version
Version 0.1.4.1 (UNKNOWN commits) X86_64
```

The following fails and should not.

```
[callen@boura ~/work/gordium/adserver feature/103144438-snap-1.0]
$ stack clean
[callen@boura ~/work/gordium/adserver feature/103144438-snap-1.0]
$ stack build
Boilerpipe-0.4.1-e9ef15bd04a5aee97899c5d0e92067fe: unregistering (local file changes)
Multiprocessing-0.1.2-7478cee629613eaadbe9e34aeff5ea2f: unregistering (local file changes)
crypto-pubkey-0.2.8-6746e013fe62668c246ed14c15438445: unregistering (missing dependencies: cryptohash)
cryptohash-0.11.6-65181ce47576341a060aa91d89501aab: unregistering (local file changes)
heist-1.0.0.0-7581df8b18fdff6ef720509a84c3f237: unregistering (local file changes)
io-streams-haproxy-1.0.0.0-6e0adb4256ccc1ceedda6f1084caaccc: unregistering (local file changes)
mysql-simple-0.2.2.6-234006630cddb8752f6f27efcc10a33d: unregistering (local file changes)
pwstore-fast-2.4.4-faf00788eb21200a99027735fca7bad7: unregistering (missing dependencies: cryptohash)
snap-core-1.0.0.0-970e0cd739853c29fd72c55614fcca79: unregistering (local file changes)
snap-server-1.0.0.0-75fb6cf08e48f2cbc2950e0b02afb4ae: unregistering (missing dependencies: io-streams-haproxy, snap-core)
tldr-config-19.6-bf43bf293923ae671ac4fcccd0371776: unregistering (missing dependencies: mysql-simple)
tldr-targeting-5.1-358b7534453322b5c7090f4ec3c9ccf2: unregistering (local file changes)
ua-parser-0.6-f727859235a18d00e0b0c82f7e549c3a: unregistering (local file changes)
Multiprocessing-0.1.2: configure
Multiprocessing-0.1.2: build
cryptohash-0.11.6: configure
cryptohash-0.11.6: build
Boilerpipe-0.4.1: configure
Multiprocessing-0.1.2: install
Boilerpipe-0.4.1: build
heist-1.0.0.0: configure
heist-1.0.0.0: build
io-streams-haproxy-1.0.0.0: configure
Boilerpipe-0.4.1: install
io-streams-haproxy-1.0.0.0: build
muttonchops-0.1.2.4: configure
muttonchops-0.1.2.4: build
io-streams-haproxy-1.0.0.0: install
mysql-simple-0.2.2.6: configure
mysql-simple-0.2.2.6: build
muttonchops-0.1.2.4: install
cryptohash-0.11.6: install
mysql-simple-0.2.2.6: install
heist-1.0.0.0: install
P             
--  While building package muttonchops-0.1.2.4 using:
      /home/callen/.stack/setup-exe-cache/setup-Simple-Cabal-1.18.1.5-x86_64-linux-ghc-7.8.4 --builddir=.stack-work/dist/x86_64-linux/Cabal-1.18.1.5/ install
    Process exited with code: ExitFailure 1
    Logs have been written to: /home/callen/work/gordium/adserver/.stack-work/logs/muttonchops-0.1.2.4.log

    Configuring muttonchops-0.1.2.4...
    Preprocessing library muttonchops-0.1.2.4...
    [1 of 1] Compiling Text.Templating.Muttonchops ( Text/Templating/Muttonchops.hs, .stack-work/dist/x86_64-linux/Cabal-1.18.1.5/build/Text/Templating/Muttonchops.o )
    crustache/src/crustache.c: In function ‘parse_mustache_name’:
    
    crustache/src/crustache.c:463:3:
         warning: implicit declaration of function ‘isalnum’ [-Wimplicit-function-declaration]
           if (!isalnum(mst->name[i]) && mst->name[i] != '_')
           ^
    crustache/src/crustache.c: In function ‘parse_mustache_name’:
    
    crustache/src/crustache.c:463:3:
         warning: implicit declaration of function ‘isalnum’ [-Wimplicit-function-declaration]
           if (!isalnum(mst->name[i]) && mst->name[i] != '_')
           ^
    crustache/src/houdini_html.c: In function ‘unescape_ent’:
    
    crustache/src/houdini_html.c:111:3:
         warning: implicit declaration of function ‘isdigit’ [-Wimplicit-function-declaration]
           if (isdigit(src[1])) {
           ^
    
    crustache/src/houdini_html.c:117:4:
         warning: implicit declaration of function ‘isxdigit’ [-Wimplicit-function-declaration]
            for (i = 2; i < size && isxdigit(src[i]); ++i)
            ^
    crustache/src/houdini_html.c: In function ‘unescape_ent’:
    
    crustache/src/houdini_html.c:111:3:
         warning: implicit declaration of function ‘isdigit’ [-Wimplicit-function-declaration]
           if (isdigit(src[1])) {
           ^
    
    crustache/src/houdini_html.c:117:4:
         warning: implicit declaration of function ‘isxdigit’ [-Wimplicit-function-declaration]
            for (i = 2; i < size && isxdigit(src[i]); ++i)
            ^
    In-place registering muttonchops-0.1.2.4...
    Installing library in
    /home/callen/work/gordium/adserver/.stack-work/install/x86_64-linux/lts-2.16/7.8.4/lib/x86_64-linux-ghc-7.8.4/muttonchops-0.1.2.4
    Installing executable(s) in
    /home/callen/work/gordium/adserver/.stack-work/install/x86_64-linux/lts-2.16/7.8.4/bin
    setup-Simple-Cabal-1.18.1.5-x86_64-linux-ghc-7.8.4:
    .stack-work/dist/x86_64-linux/Cabal-1.18.1.5/build/muttonchops-test/muttonchops-test:
    does not exist
rogr%
```

Note that muttonchops builds fine when I build it by itself:

```
Linking .stack-work/dist/x86_64-linux/Cabal-1.18.1.5/build/muttonchops-test/muttonchops-test ...
muttonchops-0.1.2.4: install
Installing library in
/home/callen/work/gordium/muttonchops/.stack-work/install/x86_64-linux/lts-2.16/7.8.4/lib/x86_64-linux-ghc-7.8.4/muttonchops-0.1.2.4
Installing executable(s) in
/home/callen/work/gordium/muttonchops/.stack-work/install/x86_64-linux/lts-2.16/7.8.4/bin
Registering muttonchops-0.1.2.4...
```

In the project that fails (adserver), muttonchops is a package dependency.

```
- location: ../muttonchops
  extra-dep: true
```

What gives?

**Comment by bitemyapp** [comment]
Disappeared after attempting `stack build` a few more times with nothing else in between. :tired_face:

**Comment by snoyberg** [comment]
Upstream bug, see https://github.com/haskell/cabal/issues/2780, https://github.com/haskell/cabal/issues/2802 and #834

**Comment by borsboom** [comment]
Closing in favour of the upstream bugs and #834.
